libs/CallRequest.js

```diff
@@ -1,150 +1,132 @@
-let callLogicFolder = __dirname+"/../../../"
-if(global.config.callLogicFolder)callLogicFolder= __dirname+"/../../../"+global.config.callLogicFolder
-console.log("Loading CallRequest Class , Call Logic folder is :",callLogicFolder)
 const clear = require("clear-module");
+const Request = require('./Request');
 
 const CallCustomParam = require("../libs/ivrAction/callParam")
 //Load Call Action
 const Say = require("../libs/ivrAction/say")
 const GoToLayer = require("../libs/ivrAction/goToLayer")
 const Dial = require("../libs/ivrAction/dial")
 
-module.exports = class CallRequest {
-    constructor(request,reply) {
-        if(request.query.reload||request.query.Reload){
-            clear(callLogicFolder+'/'+request.params.CallLogic )
-        }
-        this.request = request;
-        this.reply = reply;
-        // Main Request IVR_LAYER_INPUT Vars
-        this.Did = null
-        this.CallerID = null
-        this.CallID = null
-        this.DTMF = ""
-        this.LayerID = null
-        this.PreviousLayerID = null
-        this.CustomDataParmList =[]
+module.exports = class CallRequest extends Request {
+    constructor(request, reply) {
+        super(request, reply);
+        if (global.config.callLogicFolder) this.modulePath = global.config.callLogicFolder;
+        this.clearActionModule();
+        this.CustomDataParmList = [];
+        // Fill fields map
+        this.requestFields = new Map([
+            ['DID', 'Did'],
+            ['CALLER_ID', 'CallerID'],
+            ['IVR_UNIQUE_ID', 'CallID'],
+            ['DTMF', 'DTMF'],
+            ['LAYER_ID', 'LayerID'],
+            ['PREVIOUS_LAYER_ID', 'PreviousLayerID'],
+        ]);
+        this.requestFields.forEach((classField, bodyField) => {
+            this[classField] = null;
+        });
         // Load var From Request
-        this.parseRequest()
+        this.requireActionModule();
+        this.parseRequest();
+    }
 
+    parseRequest() {
+        let self = this;
         try {
-            this.callLogic = require( callLogicFolder+'/'+request.params.CallLogic  );
-        }
-        catch( e ) {
-            if ( e.code === 'MODULE_NOT_FOUND' ) {
-                // The module hasn't been found
-                console.error("MODULE_NOT_FOUND "+ request.params.CallLogic  ,e)
+            if (this.request.body.DATA) {
+                this.parseRequestToObject(this.request.body.DATA);
+
+                //Parsing CUSTOM_DATA
```

Comment:
instead  comment you can use the function
`this.parseCustomData`

And you can add it to main class